prov/rxm/src/rxm_cq.c

```diff
@@ -437,6 +437,35 @@ static ssize_t rxm_lmt_send_ack(struct rxm_rx_buf *rx_buf)
 	return ret;
 }
 
+static ssize_t rxm_lmt_send_ack_fast(struct rxm_rx_buf *rx_buf)
+{
+	struct rxm_pkt pkt;
+	ssize_t ret;
+
+	assert(rx_buf->conn);
+
+	pkt.hdr.op		= ofi_op_msg;
+	pkt.hdr.version		= OFI_OP_VERSION;
+	pkt.ctrl_hdr.version	= OFI_CTRL_VERSION;
+	pkt.ctrl_hdr.type	= ofi_ctrl_ack;
+	pkt.ctrl_hdr.conn_id 	= rx_buf->conn->handle.remote_key;
+	pkt.ctrl_hdr.msg_id 	= rx_buf->pkt.ctrl_hdr.msg_id;
+
+	ret = fi_inject(rx_buf->conn->msg_ep, &pkt, sizeof(pkt), 0);
+	if (OFI_UNLIKELY(ret)) {
+		FI_DBG(&rxm_prov, FI_LOG_EP_DATA,
+		       "fi_inject(ack pkt) for MSG provider failed\n");
+		return ret;
+	}
+
+	RXM_LOG_STATE(FI_LOG_CQ, rx_buf->pkt, RXM_LMT_ACK_SENT, RXM_LMT_FINISH);
+	rx_buf->hdr.state = RXM_LMT_FINISH;
+
+	if (!rx_buf->ep->rxm_mr_local)
+		rxm_ep_msg_mr_closev(rx_buf->mr, RXM_IOV_LIMIT);
+	return rxm_finish_recv(rx_buf);
```

Comment:
Can we move the above code into a function? We have the same code in rxm_cq_handle_comp for RXM_LMT_ACK_SENT state.

Reply:
Sure. Thanks
Done